pkg/splunk/client/azureblobclient.go

```diff
@@ -16,90 +16,185 @@
 package client
 
 import (
+	"bytes"
 	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
 	"encoding/json"
+	"encoding/xml"
+	"errors"
 	"fmt"
+	"io/ioutil"
 	"net/http"
-	"os"
+	"net/url"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
 
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
 // blank assignment to verify that AzureBlobClient implements BlobClient
 var _ RemoteDataClient = &AzureBlobClient{}
 
-// DownloadOptions ...Options for downloading the apps
-type DownloadOptions struct {
-	MaxChunkSize int // chunk size in gb
-	Timeout      int //timeout in seconds
+// AzureBlobClient is a client to implement Azure Blob specific APIs
+type AzureBlobClient struct {
+	BucketName         string
+	StorageAccountName string
+	SecretAccessKey    string
+	Prefix             string
+	StartAfter         string
+	Endpoint           string
+	HttpClient         SplunkHTTPClient
 }
 
-// Config contains configuration for httpclient
-type Config struct {
-	MaxRetries int         // max number of retries for the rest calls
-	HTTPClient http.Client // the http client to use for calling the rest end points
+// Structs representing http response used to unmarshal
+type ContainerProperties struct {
+	CreationTime  string `xml:"Creation-Time"`
+	LastModified  string `xml:"Last-Modified"`
+	ETag          string `xml:"Etag"`
+	ContentLength string `xml:"Content-Length"`
 }
 
-// ListBlobsOptions contains options for listing the apps
-type ListBlobsOptions struct {
-	Prefix      string // prefix of the files to list, useful to filter apps based on the path they belong to
-	MaxEnteries int    // total number of files to return in one call
+type MyBlob struct {
```

Comment:
`MyBlob`? This is the details for each individual blob to connect to, correct?

Reply:
That's correct. Do we need a better name? :)

Reply:
Just sounds very "example code" but thats a super nit so I'm fine with it.

Reply:
Nit: Agree with Jeff. we can use a better. name.

Reply:
Fixed